compute_endpoint/globus_compute_endpoint/endpoint/endpoint_manager.py

```diff
@@ -366,6 +379,32 @@ def _install_signal_handlers(self):
 
         signal.signal(signal.SIGCHLD, self.set_child_died)
 
+    def send_heartbeat(self, shutting_down=False) -> Future[None]:
```

Comment:
I guess there's a lot of logic in interchange.py that isn't necessary, and to consolidate the 'send_heartbeat' logic for both SU and MU endpoints is more trouble than it's worth?

Reply:
That was my thinking at this time, yes.  In short, the heartbeat logic is somewhat entrenched inside of the interchange in a way that makes it hard to pull out at this time without a (much) larger refactor than I wanted to do.  Someone else may see how to do it, but I do not at this time.

I think there are a number of interchange refactors (and internal redesigns) that we need to implement first, after which consolidating these two sets of heartbeat logic will likely be much clearer if not trivial.